src/generate/api.ts

```diff
@@ -5,30 +5,56 @@
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
 import { model } from "amf-client-js";
-import _ from "lodash";
 
 import {
   resolveApiModel,
   parseRamlFile,
   getAllDataTypes
 } from "../common/parser";
 
+import Name from "./name";
+
 /**
  * An API represented as an AMF model. Includes the extracted data types
  * defined in the spec. Common transformations of the name are cached for
  * reference in templates and for use in assembling file paths.
  */
-export class Api {
-  dataTypes: model.domain.CustomDomainProperty[];
-  model: model.document.Document;
-  name: {
-    original: string;
-    kebabCase: string;
-    lowerCamelCase: string;
-    snakeCase: string;
-    upperCamelCase: string;
-  };
-  path: string;
+export default class Api {
```

Comment:
I was debating whether to use default on the class. Default exports have maintainability  issues

https://basarat.gitbook.io/typescript/main-1/defaultisbad

Reply:
TypeScript convention (according to that Udemy course) is to always use named exports.